common/src/main/java/org/wso2/testgrid/common/util/S3StorageUtil.java

```diff
@@ -50,13 +65,54 @@ public static String getS3BucketURL() {
         return String.join("/", TestGridConstants.AMAZON_S3_URL, s3BucketName);
     }
 
+    public static void deleteTestPlan(TestPlan testPlan) {
```

Comment:
```suggestion
    public static boolean deleteTestPlan(TestPlan testPlan) {
```

I think you need to return the deletion success/failure status.